MediaWikiKit/MediaWikiKit/MWKLocationSearchResult.h

```diff
@@ -2,10 +2,41 @@
 #import "MWKSearchResult.h"
 @import CoreLocation;
 
+/**
+ *  Response object model for search results which have geocoordinates.
+ *
+ *  @warning This object only supports deserialization <b>from</b> JSON, not serialization <b>to</b> JSON.
+ */
 @interface MWKLocationSearchResult : MWKSearchResult<MTLJSONSerializing>
 
+/**
+ *  Location serialized from the first set of coordinates in the response.
+ */
 @property (nonatomic, strong, readonly) CLLocation* location;
 
+/**
+ *  Number of meters between the receiver and the coordinate parameters of the originating search.
+ */
 @property (nonatomic, assign, readonly) CLLocationDistance distanceFromQueryCoordinates;
 
+/**
+ *  @name Automatically Updating Properties
+ *
+ *  These properties are set automatically by @c WMFNearbyViewModel in response to location & heading updates.
+ *
+ *  @see -[WMFNearbyViewModel autoUpdateResultAtIndex:]
+ */
+
+/**
+ *  Distance in meters from the user's current location.
+ */
+@property (nonatomic, assign) CLLocationDistance distanceFromUser;
```

Comment:
I had considered doing something along the lines of what you're suggesting, but nothing immediately came to mind.  I'll mull this over a bit more, but I'm not sure if the extra indirection is worth it for the "pureness" of a 100% immutable search result.  

> Also this is conflating the user position (which is changing) with a model object that represents just a search result (which is static).

What exactly do you think is confusing about the current properties?  I thought `distanceFromQueryCoordinates` and `distanceFromUser` were pretty distinct.

> The nearby view model makes more sense as that is where user position and search results are actually used to calculate `distanceFromUser` and `bearingToLocation`.

It also already has a reference to the search results, so IMHO it was intuitive to also manage updating their properties in response to events It was already responding to.